lib/engine/game/g_1880/step/selection_auction.rb

```diff
@@ -48,10 +49,21 @@ def all_passed!
           end
 
           def post_win_bid(_winner, _company)
+            post_win
+          end
+
+          def post_win(all_passed = false)
             @round.goto_entity!(@auction_triggerer)
             entities.each(&:unpass!)
-            next_entity!
-            @auction_triggerer = current_entity
+
+            if all_passed
+              next_idx = (@round.entity_index + 1) % entities.size
+              @auction_triggerer = entities[next_idx]
+            else
+              next_entity!
+              @auction_triggerer = current_entity
+            end
```

Comment:
What's going on here? I thought all'd you have to do is set @auction_triggerer to to current_entity and then call post_win_bid. Do you really need all this complexity?

Reply:
if player_1 auto wins this by all_passing, then the next auction will be started by player_3, and not player_2. 

next_entity is called twice, once in post_win_bid, and once in in process_pass https://github.com/tobymao/18xx/blob/1dd6528a098872c003943bd4dd6506e26c99f482/lib/engine/step/selection_auction.rb#L47 

I could add a round/step param all_passed_win, to avoid the next! in process_pass, would that be cleaner?

Reply:
The code you have to work with isn't great. Nothing obvious comes to mind; however, compensating for how process_pass is implemented in your post_win implementation is very brittle code.

Reply:
added all_passed_win, is this better?